### DISCUSSION TITLE
Nginx deafult page

### DISCUSSION THREAD

**Discussion by Zordon1337** [osuAkatsuki/bancho.py#209] [General]

**Comment by lenforiee** [comment]
officially supported (-devserver) method? no, but you can create self signed certificates and use hosts file method, that’s the only way you can access the server without domain

**Comment by mayacopeland** [comment]
The hosts method will cause a ton of notification spam though, as peppy doesn't like anyone to use that method anymore. See
![screenshot004](https://user-images.githubusercontent.com/58884617/160428143-d4f42d16-aecf-45e5-abb5-0def3d1c9005.jpg)
:

**Comment by mayacopeland** [comment]
Of course, the hosts method *works* but expect issues.

**Comment by KotRikD** [comment]
just register free tk domain bro

**Comment by Zordon1337** [comment]
> officially supported (-devserver) method? no, but you can create self signed certificates and use hosts file method, that’s the only way you can access the server without domain

idk how to create self signed certificate

**Comment by lenforiee** [comment]
then just use solution in @KotRikD comment, you can get free 1 year tk domain

**Comment by Zordon1337** [comment]
Challenge failed for domain custombancho.ml
dns-01 challenge for custombancho.ml
Cleaning up challenges
Some challenges have failed.

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: custombancho.ml
   Type:   dns
   Detail: DNS problem: NXDOMAIN looking up TXT for
   _acme-challenge.custombancho.ml - check that a DNS record exists
   for this domain
   
   HELP

**Comment by yo-ru** [comment]
> Challenge failed for domain custombancho.ml
> 
> dns-01 challenge for custombancho.ml
> 
> Cleaning up challenges
> 
> Some challenges have failed.
> 
> 
> 
> IMPORTANT NOTES:
> 
>  - The following errors were reported by the server:
> 
> 
> 
>    Domain: custombancho.ml
> 
>    Type:   dns
> 
>    Detail: DNS problem: NXDOMAIN looking up TXT for
> 
>    _acme-challenge.custombancho.ml - check that a DNS record exists
> 
>    for this domain
> 
>    
> 
>    HELP
> 
> 

You need to register a domain at a registrar like namecheap. You can also register a .tk domain for free. You then need to generate a certificate. In your attempt above it asks you to add a DNS entry to ensure that you have authority over the domain you entered.

Let me know if you need anymore help,
Yoru.

You may add me on discord (Yoru#9197) if you need any more assistance.

**Comment by Zordon1337** [comment]
> > Challenge failed for domain custombancho.ml
> > dns-01 challenge for custombancho.ml
> > Cleaning up challenges
> > Some challenges have failed.
> > IMPORTANT NOTES:
> > 
> > * The following errors were reported by the server:
> >   Domain: custombancho.ml
> >   Type:   dns
> >   Detail: DNS problem: NXDOMAIN looking up TXT for
> >   _acme-challenge.custombancho.ml - check that a DNS record exists
> >   for this domain
> >   HELP
> 
> You need to register a domain at a registrar like namecheap. You can also register a .tk domain for free. You then need to generate a certificate. In your attempt above it asks you to add a DNS entry to ensure that you have authority over the domain you entered.
> 
> Let me know if you need anymore help, Yoru.
> 
> You may add me on discord (Yoru#9197) if you need any more assistance.

i changed the domain .tk  and now i have other error 
Challenge failed for domain coolbancho.tk
dns-01 challenge for coolbancho.tk
Cleaning up challenges
Some challenges have failed.

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: coolbancho.tk
   Type:   dns
   Detail: DNS problem: NXDOMAIN looking up TXT for
   _acme-challenge.coolbancho.tk - check that a DNS record exists for
   this domain

**Comment by Zordon1337** [comment]
when i add "_acme-challenge.coolbancho.tk" to name label and add code to target label and click save changes. Name changes from "_acme-challenge.coolbancho.tk" to "_ACME-CHALLENGE"

**Comment by yo-ru** [comment]
> when i add "_acme-challenge.coolbancho.tk" to name label and add code to target label and click save changes. Name changes from "_acme-challenge.coolbancho.tk" to "_ACME-CHALLENGE"

It is supposed to be a TXT record.

**Comment by Zordon1337** [comment]
i changed it to TXT record

**Comment by Zordon1337** [comment]
it seems like freenon is blocking domain name in name label

**Comment by 7ez** [comment]
transfer freenom to cloudflare lol

**Comment by Zordon1337** [comment]
> transfer freenom to cloudflare lol

BUT HOW

**Comment by Zordon1337** [comment]
help i get this when i try to transfer to cloudflare
![image](https://user-images.githubusercontent.com/65111609/160483291-003c8c7d-91c1-4801-bd3f-3233de367843.png)

**Reply by 7ez** [reply]
https://google.com

**Reply by Zordon1337** [reply]
i can't find anything

**Reply by 7ez** [reply]
clearly ur search query is wrong

**Comment by Zordon1337** [comment]
cloudflare still blocking domain name in dns record

**Reply by Zordon1337** [reply]
it finally worked

**Comment by Zordon1337** [comment]
cert generated
ngnix conf edited
but i can't connect to server

**Reply by 7ez** [reply]
send domain link

**Reply by Zordon1337** [reply]
> send domain link

http://www.zbancho.ml/

**Reply by 7ez** [reply]
> > send domain link
> 
> http://www.zbancho.ml/
> 

u didnt even make subdomains that are required

**Reply by Zordon1337** [reply]
> > > send domain link
> > 
> > 
> > http://www.zbancho.ml/
> 
> u didnt even make subdomains that are required

how to do sub domains (i know i'm dumb)

**Reply by Zordon1337** [reply]
what subdomains i must create

**Reply by lokillo-0** [reply]
> 

I must assume that you are using cloudflare you must put the subdomains in this way:
Put the first two as type A
1.- The main domain must point to the IP address of the server
2.- *www.mydomain.com* also pointing to the server IP
3.- (These are for each subdomain and are CNAME type) a, assets, b, c, ce, c4, c5, c6, i, osu, www

**Reply by Zordon1337** [reply]
"first two" you mean a, assets?

**Reply by lokillo-0** [reply]
No, I mean the first two that you put the main domain that would be "@" and the other "www" and you put them as subdomain type A

**Reply by Zordon1337** [reply]
for example www.zbancho.tk, @.zbancho.tk
?

**Reply by lokillo-0** [reply]
It would be something like this
![Screenshot_20220329-043503~2](https://user-images.githubusercontent.com/19214569/160593676-ba998b69-91b7-40fc-a27b-fd77d5121e30.png)

**Reply by Zordon1337** [reply]
ok now i understand but server ip = my ip?

**Reply by lokillo-0** [reply]
This
![Screenshot_20220329-044028](https://user-images.githubusercontent.com/19214569/160594302-f260ff93-1997-416a-a943-bca5c043c2a1.png)

**Reply by lokillo-0** [reply]
I want to assume that if it works on your computer, you would only have to open the ports of your router (I don't remember which ones were lmao) (and if you plan to play with other people) but I consider a better option to rent a cheap vps if many people are not going to play (10 or 15 at most)

**Reply by Zordon1337** [reply]
i'm thinking about a buying cheap vps

**Reply by Zordon1337** [reply]
does vps setup look the same as wsl?

**Reply by 7ez** [reply]
> > 
> 
> I must assume that you are using cloudflare you must put the subdomains in this way:
> Put the first two as type A
> 1.- The main domain must point to the IP address of the server
> 2.- *www.mydomain.com* also pointing to the server IP
> 3.- (These are for each subdomain and are CNAME type) a, assets, b, c, ce, c4, c5, c6, i, osu, www

c5 and c6 are useless + nobody uses www these days anyway

**Reply by lokillo-0** [reply]
In my server I have it in contabo, it cost me 7 usd and it has 4 Vcpu, 8GB of ram and you can choose between 50GB nvme SSD or 200GB traditional SSD

**Reply by Zordon1337** [reply]
is it enough for 2 players? 
![image](https://user-images.githubusercontent.com/65111609/160596632-91b9649e-0168-4b9f-ba49-864922fe8c5d.png)

**Reply by 7ez** [reply]
more than enough

**Reply by lokillo-0** [reply]
> is it enough for 2 players? ![image](https://user-images.githubusercontent.com/65111609/160596632-91b9649e-0168-4b9f-ba49-864922fe8c5d.png)

Ye

**Reply by Zordon1337** [reply]
can i will use zbancho.ml domain on vps?

**Reply by lokillo-0** [reply]
> can i will use zbancho.ml domain on vps?

yes, just as I said the first 2 subdomains (those of type A) must point to the IP of your server

**Reply by 7ez** [reply]
> can i will use zbancho.ml domain on vps?
> 

can I use* lol

**Reply by Zordon1337** [reply]
> It would be something like this ![Screenshot_20220329-043503~2](https://user-images.githubusercontent.com/19214569/160593676-ba998b69-91b7-40fc-a27b-fd77d5121e30.png)

![image](https://user-images.githubusercontent.com/65111609/160599395-5c5de80d-47be-481a-8552-bf651bfbce04.png)
it should be like this?

**Reply by Zordon1337** [reply]
idk why but for example www.zbancho.ml is just www

**Reply by 7ez** [reply]
> idk why but for example www.zbancho.ml is just www

thats how its supposed to look like

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160600393-0ab1da16-cb2f-45f7-b770-dead664a9bbb.png)

**Reply by lokillo-0** [reply]
> ![image](https://user-images.githubusercontent.com/65111609/160600393-0ab1da16-cb2f-45f7-b770-dead664a9bbb.png)

Try c.zbancho.ml

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160600675-2aeb2d4f-3281-4646-9623-f6daafcc66e3.png)

**Reply by lokillo-0** [reply]
if you did the whole tutorial right? included change nginx files

**Reply by Zordon1337** [reply]
yes

**Reply by Zordon1337** [reply]
i'm gonna try 
sudo nginx -s reload

**Reply by Zordon1337** [reply]
maybe i know why 
![image](https://user-images.githubusercontent.com/65111609/160601140-cca06602-3b6f-407a-ad09-88e2483a7539.png)

**Reply by Zordon1337** [reply]
> maybe i know why 
> ![image](https://user-images.githubusercontent.com/65111609/160601140-cca06602-3b6f-407a-ad09-88e2483a7539.png)
> 

Any idea how to fix?

**Reply by 7ez** [reply]
u can't host a server for someone that's outside ur network psure

**Reply by KotRikD** [reply]
`sudo nginx -t` ? This will show config errors if they exists. And even tho try use `systemctl restart nginx.service` instead of direct restart

**Reply by Zordon1337** [reply]
> `sudo nginx -t` ? This will show config errors if they exists. And even tho try use `systemctl restart nginx.service` instead of direct restart

![image](https://user-images.githubusercontent.com/65111609/160605178-54db24f8-0aa3-4e71-a8ef-507cc08a919e.png)

![image](https://user-images.githubusercontent.com/65111609/160605256-3d6e4d4f-1745-48ba-ada6-ccc8e712d315.png)

**Reply by KotRikD** [reply]
`sudo service nginx start`?

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160605621-566568f3-b62a-4644-afec-ef9f5075c39c.png)

**Reply by 7ez** [reply]
just do sudo nginx and if it tells you listening on port failed then it means it's running already

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160611722-f7e2dc3a-48c4-47be-a7c9-bbc9e657606f.png)

**Reply by Zordon1337** [reply]
ngnix not installed?

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160612282-1ba39f6d-f562-4ec0-ab7b-28e06fca507b.png)

**Reply by 7ez** [reply]
u cant spell nginx kek

**Reply by KotRikD** [reply]
bro just restart wsl idk xd. `wsl --shutdown`

**Reply by Zordon1337** [reply]
i'm using now vps

**Reply by Zordon1337** [reply]
help 
![image](https://user-images.githubusercontent.com/65111609/160626770-ddbfa83f-a50f-4213-a5c0-e6cc3c8d26d6.png)

**Reply by 7ez** [reply]
python3 -m pip install requests 🤦‍♂️

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160630457-0753ddb8-c6de-4522-ae61-067a110ef331.png)

**Reply by 7ez** [reply]
...

**Reply by Zordon1337** [reply]
because python3 -m pip not working

**Reply by 7ez** [reply]
then install pip for it

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160631530-b70566d4-fb8e-4a08-9713-8cc87e4e480d.png)

**Reply by Zordon1337** [reply]
pip3 worked

**Comment by mayacopeland** [comment]
How TF do i unsubscribe from this i have 50 emails from it

**Reply by KotRikD** [reply]
Just press unsubscribe at top of the topic

**Reply by mayacopeland** [reply]
oh i never noticed

**Comment by Zordon1337** [comment]
ok. so i sucessfully started bancho.py on vps but i still can't connect and when i go on zbancho.ml or c.zbancho.ml i get ngnix page Help?

**Reply by Zordon1337** [reply]
i can't even connect on b20151128.2 fallback

**Reply by KotRikD** [reply]
1. enter in your windows cmd `tracert c.zbancho.ml` and check which host you connecting.
2. dont use fallback version, i really thing that bancho.py will not work with that odd and old version

**Reply by 7ez** [reply]
why are u using fallback

**Reply by 7ez** [reply]
it's outdated and won't let you submit

**Reply by 7ez** [reply]
and probably restrict you also

**Reply by Zordon1337** [reply]
> 1. enter in your windows cmd `tracert c.zbancho.ml` and check which host you connecting.
> 2. dont use fallback version, i really thing that bancho.py will not work with that odd and old version

![image](https://user-images.githubusercontent.com/65111609/160648948-6122a57e-6ba0-4170-8cb3-c2bfb7de7791.png)

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160649180-e33ebff4-278b-439e-b264-9d238663ab2e.png)

**Comment by Zordon1337** [comment]
i think the problem is that ngnix is connecting to zbancho.ml but bancho.py can't connect with zbancho.ml

**Reply by 7ez** [reply]
you fucked up nginx config

**Reply by Zordon1337** [reply]
i'm gonna try fix

**Reply by Zordon1337** [reply]
> you fucked up nginx config

ngnix config is in bancho.py/etc/ngnix.conf ?

**Reply by Zordon1337** [reply]
if yes i know what is wrong

**Reply by 7ez** [reply]
it is and its nginx not ngnix

**Reply by Zordon1337** [reply]
# NOTE: if you wish to only connect using fallback, you can
# remove all ssl related content from the configuration.

upstream bancho {
	server unix:/tmp/bancho.sock fail_timeout=0;
}

# c[e4]?.ppy.sh is used for bancho
# osu.ppy.sh is used for /web, /api, etc.
# a.ppy.sh is used for osu! avatars

server {
	listen 443 ssl;

	# XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
	server_name ~^(?:c[e4]?|osu|a|b)\.zbancho\.ml$;

	# XXX: you'll need to edit this to match your certificate and key
	ssl_certificate     /etc/letsencrypt/live/zbancho.ml/fullchain.pem;
	ssl_certificate_key /etc/letsencrypt/live/zbancho.ml/privkey.pem;

	# TODO: further ssl configuration
	ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

	location / {
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header X-Real-IP  $remote_addr;
		proxy_set_header Host $http_host;
		proxy_redirect off;
		proxy_pass http://bancho;
	}
}

server {
	listen 443 ssl;

	# XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
	server_name assets.zbancho.ml;

	# XXX: you'll need to edit this to match your certificate and key
	ssl_certificate     /etc/letsencrypt/live/zbancho.ml/fullchain.pem;
	ssl_certificate_key /etc/letsencrypt/live/zbancho.ml/privkey.pem;

	# TODO: further ssl configuration
	ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

	location / {
		default_type image/png;
		# XXX: you'll need to edit this to match your bancho.py .data/assets directory
		root /home/root/programming/bancho.py/.data/assets/;
	}
}



Is it good now?

**Reply by Zordon1337** [reply]
still only nginx page i even restarted vps, bancho.py, nginx

**Reply by 7ez** [reply]
do u have the config in /etc/nginx/sites-enabled

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160664381-9f0a8280-8398-477e-baf1-82224dcda924.png)

**Reply by Zordon1337** [reply]
etc folder disappeared from bancho.py folder

**Reply by 7ez** [reply]
remove default

**Reply by Zordon1337** [reply]
done. what should i do now

**Reply by Zordon1337** [reply]
restart?

**Reply by Zordon1337** [reply]
![image](https://user-images.githubusercontent.com/65111609/160665422-e0ecbdc5-555e-482a-9d58-2332481eb105.png)

**Reply by 7ez** [reply]
something def wrong with nginx conf

**Reply by Zordon1337** [reply]
but where is nginx conf because the etc folder in bancho.py disappered

**Reply by Zordon1337** [reply]
root/etc/nginx/nginx.conf ?

**Reply by Zordon1337** [reply]
help

root@Zbancho:~ sudo service nginx start
Job for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xe" for details.
root@Zbancho:~ systemctl status nginx.service
● nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Tue 2022-03-29 17:40:19 UTC; 28s ago
       Docs: man:nginx(8)
    Process: 981 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=1/FAILURE)

Mar 29 17:40:19 Zbancho systemd[1]: Starting A high performance web server and a reverse proxy server...
Mar 29 17:40:19 Zbancho nginx[981]: nginx: [emerg] "upstream" directive is not allowed here in /etc/nginx/nginx.conf:2
Mar 29 17:40:19 Zbancho nginx[981]: nginx: configuration file /etc/nginx/nginx.conf test failed
Mar 29 17:40:19 Zbancho systemd[1]: nginx.service: Control process exited, code=exited, status=1/FAILURE
Mar 29 17:40:19 Zbancho systemd[1]: nginx.service: Failed with result 'exit-code'.
Mar 29 17:40:19 Zbancho systemd[1]: Failed to start A high performance web server and a reverse proxy server.

**Reply by Zordon1337** [reply]
nvrmind conf fault

**Reply by Zordon1337** [reply]
> 7ez

i don't see anything wrong here 
![image](https://user-images.githubusercontent.com/65111609/160675252-2dcb9f73-e95a-4c07-b43c-bcdf203528b9.png)

**Comment by Zordon1337** [comment]
bancho.py recived request

**Comment by Zordon1337** [comment]
# here is more details (maybe this will help)

# Nginx config
![image](https://user-images.githubusercontent.com/65111609/160691692-a1a75620-47fe-4482-a2b0-d420f73269da.png)

# bancho.py config

![image](https://user-images.githubusercontent.com/65111609/160692137-dfaa1203-a85c-4ed0-b98d-c27322a9c8f3.png)

# bancho.py console logs
# Note: I Can't Run main.py without sudo because my vps is logged as root

![image](https://user-images.githubusercontent.com/65111609/160692849-7b00f8b2-5eca-47a3-a6d4-2a3dde464b06.png)


# ----------the problems----------



# i stuck on nginx page doesn't matter if i'm on c.zbancho.ml or on zbancho.ml
![image](https://user-images.githubusercontent.com/65111609/160693601-3ae5f5d6-a7ee-4b78-a009-ce73e8107cbe.png)


# 2nd problem can't connect from osu 
# NOT MY SCREENSHOT
![image](https://user-images.githubusercontent.com/65111609/160693873-4a6535b1-130b-4427-af56-01665970bbb7.png)

**Comment by Zordon1337** [comment]
okay, sorry for being dumb asf 2 years ago
